bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,329 @@
+    logger.trace("Received command {} on channel {}", command, channelUID.getId());
+    try {
+      PJLinkDevice device = getDevice();
+      switch (channelUID.getId()) {
+        case CHANNEL_POWER:
+          logger.trace("Received power command" + command);
+          if (command == OnOffType.ON) {
+            device.powerOn();
+          } else if (command == OnOffType.OFF) {
+            device.powerOff();
+          } else if (command == RefreshType.REFRESH) {
+            updateState(PJLinkDeviceBindingConstants.CHANNEL_POWER,
+                PowerQueryResponseValue.POWER_ON.equals(device.getPowerStatus().getResult()) ? OnOffType.ON
+                    : OnOffType.OFF);
+          } else {
+            logger.debug("Received unknown power command" + command);
+          }
+          break;
+        case CHANNEL_INPUT:
+          if (command == RefreshType.REFRESH) {
+            StringType input = new StringType(device.getInputStatus().getResult().getValue());
+            updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT, input);
+          } else if (command instanceof StringType) {
+            logger.trace("Received input command" + command);
+            Input input = new Input(((StringType) command).toString());
+            device.setInput(input);
+          } else {
+            logger.debug("Received unknown channel command" + command);
+          }
+          break;
+        case CHANNEL_AUDIO_MUTE:
+        case CHANNEL_VIDEO_MUTE:
+          boolean isAudioMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE);
+          boolean isVideoMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE);
+          if (isVideoMute || isAudioMute) {
+            if (command == RefreshType.REFRESH) {
+              // refresh both video and audio mute, as it's one request
+              MuteQueryResponseValue muteStatus = device.getMuteStatus();
+              updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                  muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+              updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                  muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+            } else {
+              if (isAudioMute) {
+                logger.trace("Received audio mute command" + command);
+                boolean muteOn = command == OnOffType.ON;
+                device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+              }
+              if (isVideoMute) {
+                logger.trace("Received video mute command" + command);
+                boolean muteOn = command == OnOffType.ON;
+                device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+              }
+            }
+          }
+      }
+
+      updateStatus(ThingStatus.ONLINE);
+      logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+    } catch (IOException e) {
```

Comment:
```suggestion
    } catch (IOException | ResponseException e) {
```

and remove the other catch clauses. simpler code is better

Reply:
Which other catch clauses should I remove? I need them to be able to issue different ThingStatusDetails.

Reply:
The `ResponseException` immediately below